Dashboard/app/js/lib/router/router.js

```diff
@@ -231,9 +227,15 @@ FLOW.Router = Ember.Router.extend({
       editSurveysAssignment: Ember.Route.extend({
         route: '/assign-surveys',
         connectOutlets(router) {
-          router
-            .get('navDevicesController')
-            .connectOutlet('editSurveyAssignment');
+          router.get('navDevicesController').connectOutlet('editSurveyAssignment');
+
+          // if editing fetch datapoints assignments,
+          // else well, no need for datapoints assignments in a new assignments
+          const assignmentId = FLOW.selectedControl.selectedSurveyAssignment.get('keyId');
+          if (assignmentId) {
+            FLOW.dataPointAssignmentControl.populate(assignmentId);
```

Comment:
I'm loading them here to be able to get each data point assignment when it's needed rather than making another API request when in the assignment. And also other checks like, when changing survey folder the datapoints assignments need to be removed, If they're not loaded it wouldn't know what to remove.

Reply:
Then maybe its enough to skip them if there is nothing to remove. and remove if there is something.  So right now you go to the assignments page, select an assignment to edit and there you see a form and a list of devices that are part of that assignment.  however, its only relevant to have information on the datapoint assignment if you actually select a specific device to look at. if you never select a device then we have spent resources and time loading up datapoint assignments which will not be used.  So I can understand the reasoning but I think it will slow down the performance unnecessarily.

Reply:
I'm loading them here to be able to get each data point assignment when it's needed rather than making another API request when in the assignment. And also other checks like, when changing survey folder the datapoints assignments need to be removed, If they're not loaded it wouldn't know what to remove.

Reply:
True, I see your point. So that means, only load datapoint related to a device when the user clicks on it.

Reply:
Yes exactly 👍